auction-server/src/state.rs

```diff
@@ -434,32 +789,15 @@ impl SimulatedBidCoreFields {
 }
 
 impl SimulatedBid {
-    pub fn get_core_fields(&self) -> SimulatedBidCoreFields {
+    pub fn get_id(&self) -> BidId {
         match self {
-            SimulatedBid::Evm(bid) => bid.core_fields.clone(),
-            SimulatedBid::Svm(bid) => bid.core_fields.clone(),
-        }
-    }
-
-    pub fn get_auction_key(&self) -> AuctionKey {
-        let core_fields = self.get_core_fields();
-        (
-            core_fields.permission_key.clone(),
-            core_fields.chain_id.clone(),
-        )
-    }
-
-    pub fn update_status(self, status: BidStatus) -> Self {
-        let mut core_fields = self.get_core_fields();
-        core_fields.status = status;
-        match self {
-            SimulatedBid::Evm(bid) => SimulatedBid::Evm(SimulatedBidEvm { core_fields, ..bid }),
-            SimulatedBid::Svm(bid) => SimulatedBid::Svm(SimulatedBidSvm { core_fields, ..bid }),
+            SimulatedBid::Evm(bid) => bid.core_fields.id,
+            SimulatedBid::Svm(bid) => bid.core_fields.id,
         }
     }
 }
 
-impl TryFrom<(models::Bid, Option<models::Auction>)> for BidStatus {
+impl TryFrom<(models::Bid, Option<models::Auction>)> for BidStatusEvm {
```

Comment:
I'd rather we use From and TryFrom traits only for conversions that are simple.
We can put these in dedicated functions inside the traits.

it's not easy to discover `let a: BidStatusEvm = (bid, auction).try_into()` is doable

Reply:
It's better to put them inside a function.
The other reason for this is you cannot get the try_from error type when implementing a try_from for a trait.